File: api/Services/HomeService.cs
using api.Services.Interfaces;

namespace api.Services;

public class HomeService : IHomeService
{
    public async Task<object> GetApiStatusAsync()
    {
        return await Task.FromResult(new 
        { 
            message = "SnapOut API is running", 
            timestamp = DateTime.UtcNow, 
            version = "1.0.0" 
        });
    }

    public async Task<object> GetHealthAsync()
    {
        // Could add actual health checks here (database, external services, etc.)
        return await Task.FromResult(new 
        { 
            status = "healthy", 
            timestamp = DateTime.UtcNow 
        });
    }

    public async Task<object> GetApiInfoAsync()
    {
        var environment = Environment.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT") ?? "Production";
        var endpoints = new[] 
        { 
            "/api/home", 
            "/api/home/health", 
            "/api/home/info", 
            "/api/users", 
            "/api/posts" 
        };

        return await Task.FromResult(new 
        { 
            api = "SnapOut API",
            environment,
            timestamp = DateTime.UtcNow,
            endpoints
        });
    }
}

File: client/src/services/homeService.ts
// src/services/homeService.ts
import api from "./api";

export interface HomeRoot {
  message: string;
  timestamp: string; // ISO UTC
  version: string;
}

export interface HomeHealth {
  status: "healthy";
  timestamp: string; // ISO UTC
}

export interface HomeInfo {
  api: string;
  environment: string;
  timestamp: string; // ISO UTC
  endpoints: string[];
}

export const homeService = {
  async root() {
    const { data } = await api.get<HomeRoot>("/api/home");
    return data;
  },

  async health() {
    const { data } = await api.get<HomeHealth>("/api/home/health");
    return data;
  },

  async info() {
    const { data } = await api.get<HomeInfo>("/api/home/info");
    return data;
  },
};